Style: address requested issues or improvements for DTS

kylebonnici opened the pull request:
~This PR removes all indents of includes and linked comments as pre request of @nordicjm in https://github.com/zephyrproject-rtos/zephyr/pull/101546#discussion_r2652420482.~

~Commits https://github.com/zephyrproject-rtos/zephyr/commit/46a02bfa31d0dc2822e57d162b0e57135535cc1a , https://github.com/zephyrproject-rtos/zephyr/commit/bd4e631b42aa0198ee953afca6d64ddc50ddd540 , 3322e2df82be9d7f8b8427a94f7e49ed9e447de7 and cb7b96bf55020ec14602e8f0ed3b28838aaf8599 ...  should be ignored as this is part of https://github.com/zephyrproject-rtos/zephyr/pull/100388 and https://github.com/zephyrproject-rtos/zephyr/pull/101116~

UPDATE:

74511c04ead9fda7319d86fd440d1f87d393356a enforces https://github.com/zephyrproject-rtos/zephyr/pull/103714
74511c04ead9fda7319d86fd440d1f87d393356a enforces, 4432ee99205e3aafe1585ab900858f48f15d0f82 addresses change requested by @nordicjm  here https://github.com/zephyrproject-rtos/zephyr/pull/101546#discussion_r2652420482
74511c04ead9fda7319d86fd440d1f87d393356a enforces, 6f4b02db098546b77eda85944f284dd288360ad0 addresses bug raised by @nordicjm  here https://github.com/zephyrproject-rtos/zephyr/pull/101619#discussion_r2652834484
74511c04ead9fda7319d86fd440d1f87d393356a enforces, 725fe938a9f210563ec11b124e76ec2f98b50a33 addresses improvement requested by @mathieuchopstm @etienne-lms here https://github.com/zephyrproject-rtos/zephyr/pull/101116#discussion_r2768280080

kylebonnici committed:
Style: Remove include indentations

All includes and linked comments should not be indented.

Signed-off-by: Kyle Bonnici <kylebonnici@hotmail.com>

nordicjm commented on dts/arm/nordic/nrf5340_cpuapp.dtsi:
```diff
@@ -65,10 +65,10 @@
-			#include "nrf5340_cpuapp_peripherals.dtsi"
+/* Common nRF5340 Application MCU
+ * peripheral description
+ */
```
this looks a bit funky but the comment is not needed so can go entirely
```suggestion
```

nordicjm commented on dts/arm/nordic/nrf91.dtsi:
```diff
@@ -43,8 +43,8 @@
 
-			/* Common nRF91 peripherals description. */
-			#include "nrf91_peripherals.dtsi"
+/* Common nRF91 peripherals description. */
```
```suggestion
```

nordicjm commented on dts/arm/nordic/nrf91ns.dtsi:
```diff
@@ -44,8 +44,8 @@
 
-			/* Common nRF91 peripherals description. */
-			#include "nrf91_peripherals.dtsi"
+/* Common nRF91 peripherals description. */
```
```suggestion
```

kylebonnici commented on scripts/dts/python-devicetree/tests/test.dts:
```diff
@@ -307,9 +307,9 @@
-				  4 0xE 0xF 1 /* 4    -> 0xE 0xF */ >;
+			ranges = <1 0xa 0xb 1 /* 1    -> 0xA 0xB */
+				  2 0xc 0xd 2 /* 2..3 -> 0xC 0xD */
+				  4 0xe 0xf 1 /* 4    -> 0xE 0xF */ >;
```
I will look at fixing that space after the comment. Thanks for pointing this out. 

This will be in an other PR  after some of the current PRs this PR is build on are merged.

kylebonnici committed:
Style: remove space between block comment and > ]

Linter not ensures that > and ] have no white spaces when these are
preceded with a block comment.

Signed-off-by: Kyle Bonnici <kylebonnici@hotmail.com>

kylebonnici committed:
Style: remove \ from dts files

Lines ending with \ are not needed unless when used in #define.

This commit removes all cases of \ where this is not needed in
devicetree files.

Signed-off-by: Kyle Bonnici <kylebonnici@hotmail.com>

kylebonnici committed:
CI: Update dts-linter to dts-linter 0.4.1

Security:
- Update dependency to address CWE-1333, CWE-407

Changed:
- `#includes` are no longer indented to match the depth of the node they
are in.

Added:
- Formatting: removes lines ending with `\` when this is not required.
- Formatting: node names are formatted to not have `0x` and `ULL`

Fixed:
- Formatting: Remove extra space between `>` or `]` when token before is
from some comment

Signed-off-by: Kyle Bonnici <kylebonnici@hotmail.com>

etienne-lms reviewed (dismissed):
LGTM. Targeting v4.5.0 merge window?

etienne-lms reviewed (approved):
Resulting changes from linter still LGTM.

fabiobaltieri commented:
@MaureenHelm would this go in during freeze? probably not... then there's no point keep rebasing it

jadonk commented on boards/beagle/pocketbeagle_2/pocketbeagle_2_m4-pinctrl.dtsi:
```diff
@@ -27,6 +27,6 @@
 	main_uart0_pins_default: main-uart0-default-pins {
 		pinmux = <K3_PINMUX(0x1c8, PIN_INPUT, MUX_MODE_0) /* (D14/A13) UART0_RXD */
-			  K3_PINMUX(0x1cc, PIN_OUTPUT, MUX_MODE_0) /* (E14/E11) UART0_TXD */ >;
+			  K3_PINMUX(0x1cc, PIN_OUTPUT, MUX_MODE_0) /* (E14/E11) UART0_TXD */>;
```
I personally prefer the space, but if the linter settings are agreed by the maintainers, I'm ok with this whitespace change.

kylebonnici replied:
This is stemming from `prop = <10 20 30>;` given that we enforce no spaces after `<`and before `>` this should also apply to comments hence the change you see in this PR.

this rule is not explicitly mentioned here https://docs.zephyrproject.org/latest/contribute/style/devicetree.html but the example does show `sector-size = <512>;` and also here https://docs.kernel.org/devicetree/bindings/dts-coding-style.html `vendor,custom-property = <2>;`

jadonk replied:
Unless we want to make comments an exception, let's roll with it.

jadonk reviewed (commented):
The `#include` makes some sense from a pre-processor perspective, thinking about the result, **BUT** this looks pretty ugly and points to fundamental issues from leaning too hard on the pre-processor.

For the most part, I'd consider this pre-processor abuse, but not egregious.

What stands out in the way this nested use is done is that the same physical resources are being specified multiple times at different locations in the device tree. I guess these two CPUs aren't running in an SMP configuration? It seems likely if there is hope to run in an SMP configuration, specifying the SoC peripherals outside of the CPU would be a better structure to the drivers, not just the device tree maintenance. 

I'm not sure it makes sense for me to put my approval on this given how small a chunk this hits on `beagle`, but I am find with the changes to `beagle` done in this series.

kylebonnici commented:
Should we get this merged in @MaureenHelm ?
